src/schemas/json/uplugin.json

```diff
@@ -0,0 +1,401 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildTargetType"
+          }
+        },
+        "Type": {
+          "description": "Usage type of module",
+          "type": "string",
+          "enum": [
+            "Runtime",
+            "RuntimeNoCommandlet",
+            "RuntimeAndProgram",
+            "CookedOnly",
+            "UncookedOnly",
+            "Developer",
+            "DeveloperTool",
+            "Editor",
+            "EditorNoCommandlet",
+            "EditorAndProgram",
+            "Program",
+            "ServerOnly",
+            "ClientOnly",
+            "ClientOnlyNoCommandlet",
+            "Max"
+          ]
+        }
+      }
+    },
+    "PluginReferenceDescriptor": {
+      "description": "Descriptor for a plugin reference.",
+      "type": "object",
+      "properties": {
+        "Enabled": {
+          "description": "Whether it should be enabled by default",
+          "type": "boolean",
+          "default": true
+        },
+        "HasExplicitPlatforms": {
+          "description": "When true, empty `SupportedTargetPlatforms` and `PlatformAllowList` are interpreted as *no platforms* with the expectation that explicit platforms will be added in plugin platform extensions",
+          "type": "boolean",
+          "default": false
+        },
+        "Optional": {
+          "description": "Whether this plugin is optional, and the game should silently ignore it not being present",
+          "type": "boolean",
+          "default": false
+        },
+        "Description": {
+          "description": "Description of the plugin for users that do not have it installed.",
+          "type": "string"
+        },
+        "MarketplaceURL": {
+          "description": "URL for this plugin on the marketplace, if the user doesn't have it installed.",
+          "type": "string"
+        },
+        "Name": {
+          "description": "Name of the plugin",
+          "type": "string"
+        },
+        "PlatformAllowList": {
+          "description": "List of allowed platforms",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "PlatformDenyList": {
+          "description": "List of disallowed platforms",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "TargetAllowList": {
+          "description": "List of allowed targets",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildTargetType"
+          }
+        },
+        "TargetConfigurationAllowList": {
+          "description": "List of allowed target configurations",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildConfiguration"
+          }
+        },
+        "TargetConfigurationDenyList": {
+          "description": "List of disallowed target configurations",
+          "type": "array",
```

Comment:
Optional fix: add `"uniqueItems": true` to make sure that there are no duplicates.

Reply:
Optional fix implemented in 262cb20dcfc106d02304efb65aa070fa0db880cc.